app/src/main/java/com/example/music_app/data/repositories/playlists/PlaylistsRepositoryImpl.kt

```diff
@@ -29,17 +42,21 @@ class PlaylistsRepositoryImpl(
             offset = offset.toString(),
             limit = limit.toString()
         )
-        val (totalSize, playlists) = with(playlistsResponse) {
-            total to items?.map(PlaylistItemResponse::toPlaylist)
+        with(playlistsResponse) {
+            val totalSize = total
+            val (playlistsEntities, playlists) =
+                items?.map(PlaylistItemResponse::toPlaylistEntity) to items?.map(PlaylistItemResponse::toPlaylist)
+            emit(
+                catchAPIErrors(
+                    isSuccess = playlists != null && totalSize != null,
+                    error = error
+                ) {
+                    database.dao.upsertListOfPlaylists(playlistsEntities ?: listOf())
+                    Ok(ListOfPlaylists(playlists = database.dao.getAllPlaylistsSortedByPinDate(limit = limit + offset).map(
```

Comment:
I would recommend to assign to some variable the playlist value and the pass it here to make it more readable